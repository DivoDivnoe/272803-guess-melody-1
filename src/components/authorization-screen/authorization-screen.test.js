import React from 'react';
import renderer from 'react-test-renderer';
import AuthorizationScreen from './authorization-screen.jsx';
import {BrowserRouter} from 'react-router-dom';

const mock = {
  mistakes: 5,
  restart: jest.fn(),
  authUserHandler: jest.fn(),
  history: {},
};

describe(`AuthorizationScreen component`, () => {
  it(`renders correctly`, () => {
    const tree = renderer.create(
        <BrowserRouter>
          <AuthorizationScreen
            restart={mock.restart}
            authUserHandler={mock.authUserHandler}
            mistakes={mock.mistakes}
            history={mock.history}
          />
        </BrowserRouter>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
