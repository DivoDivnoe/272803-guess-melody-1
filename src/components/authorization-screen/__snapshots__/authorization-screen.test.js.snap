// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthorizationScreen component renders correctly 1`] = `
<section
  className="login"
>
  <div
    className="login__logo"
  >
    <img
      alt="Угадай мелодию"
      height="83"
      src="img/melody-logo.png"
      width="186"
    />
  </div>
  <h2
    className="login__title"
  >
    Вы настоящий меломан!
  </h2>
  <p
    className="login__total"
  >
    За 3 минуты и 25 секунд вы набрали 12 баллов (8 быстрых), совершив 
    5
     ошибки
  </p>
  <p
    className="login__text"
  >
    Хотите сравнить свой результат с предыдущими попытками? Представтесь!
  </p>
  <form
    action=""
    className="login__form"
    onSubmit={[Function]}
  >
    <p
      className="login__field"
    >
      <label
        className="login__label"
        htmlFor="name"
      >
        Логин
      </label>
      <input
        className="login__input"
        id="name"
        name="name"
        required={true}
        type="text"
      />
    </p>
    <p
      className="login__field"
    >
      <label
        className="login__label"
        htmlFor="password"
      >
        Пароль
      </label>
      <input
        className="login__input"
        id="password"
        name="password"
        required={true}
        type="text"
      />
      <span
        className="login__error"
      >
        Неверный пароль
      </span>
    </p>
    <button
      className="login__button button"
      type="submit"
    >
      Войти
    </button>
  </form>
  <a
    className="replay"
    href="/"
    onClick={[Function]}
  >
    Сыграть ещё раз
  </a>
</section>
`;
